src/nft/components/collection/CollectionNfts.tsx

```diff
@@ -43,18 +190,37 @@ export const CollectionNfts = ({ contractAddress }: CollectionNftsProps) => {
         contractAddress,
         markets,
         notForSale: !buyNow,
-        price: {
-          low: debouncedMinPrice,
-          high: debouncedMaxPrice,
-          symbol: 'ETH',
-        },
+        sortBy,
+        searchByNameText,
+        debouncedMinPrice,
+        debouncedMaxPrice,
+        searchText: searchByNameText,
       },
     ],
     async ({ pageParam = 0 }) => {
+      let sort = null
+      switch (sortBy) {
+        case SortBy.HighToLow: {
+          sort = { currentEthPrice: 'desc' }
+          break
+        }
+        case SortBy.RareToCommon: {
+          sort = { 'rarity.providers.0.rank': 1 }
+          break
+        }
+        case SortBy.CommonToRare: {
+          sort = { 'rarity.providers.0.rank': -1 }
+          break
+        }
+        default:
+      }
+
       return await AssetsFetcher({
         contractAddress,
+        sort: sort ?? undefined,
```

Comment:
couldn't you just declare sort where it will start as undefined instead of assigning it to null?